bundle/terranova/tnresources/database_instance.go

```diff
@@ -0,0 +1,97 @@
+package tnresources
+
+import (
+	"context"
+	"time"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/database"
+)
+
+type ResourceDatabaseInstance struct {
+	client *databricks.WorkspaceClient
+	config database.DatabaseInstance
+}
+
+func (d ResourceDatabaseInstance) Config() any {
+	return d.config
+}
+
+func (d ResourceDatabaseInstance) DoCreate(ctx context.Context) (string, error) {
+	response, err := d.client.Database.CreateDatabaseInstance(ctx, database.CreateDatabaseInstanceRequest{
```

Comment:
`CreateDatabaseInstance`  seems to return a waiter (https://github.com/databricks/databricks-sdk-go/blob/b6d5a0804666280e8eb7d22250d3b3c8f163eaf2/service/database/api.go#L249)

Can you store it in `ResourceDatabaseInstance` and then use it in `WaitAfterCreate`? 
Then you don't need all the custom timer logic there

Reply:
thanks for pointing it out! changed in [476a399](https://github.com/databricks/cli/pull/3283/commits/476a399e06c217606a0ac104e2f0800f87ecccb0)